src/noshow/api/app_helpers.py

```diff
@@ -69,3 +70,50 @@ def fix_outdated_appointments(
             apiprediction.active = False
             session.merge(apiprediction)
             session.commit()
+
+
+def create_treatment_groups(predictions: pd.DataFrame) -> pd.DataFrame:
+    """
+    Create treatment groups based on predictions.
+
+    Parameters
+    ----------
+    predictions : pd.DataFrame
+        DataFrame containing predictions.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame with treatment group assignments.
+
+    Raises
+    ------
+    ValueError
+        If the predictions DataFrame is empty.
+    """
+    if predictions.empty:
+        raise ValueError("The predictions DataFrame is empty.")
+
+    # Create prediction score bins using quantile bins, for example 10
+    predictions["score_bin"] = pd.qcut(predictions["prediction"], q=10)
+
+    # sort predictions by hoofdagenda and score_bin
+    predictions = predictions.sort_values(["prediction"])
+
+    # Pre-process data to handle duplicates
```

Comment:
This comment also doens't explain why, a better comment would be: `only keep top prediction per patient for treatment/control split`